### ISSUE TITLE
Regarding the issue of saving models

### ISSUE THREAD

**Issue by wzhiyuan2016** [luxiaolili/IDM-VTON-train#15]
if global_step % args.checkpointing_epoch == 0:
            if accelerator.is_main_process:
                # _before_ saving state, check if this save would set us over the `checkpoints_total_limit`
                unwrapped_unet = accelerator.unwrap_model(
                    unet, keep_fp32_wrapper=True
                )
                pipeline = TryonPipeline.from_pretrained(
                    args.pretrained_model_name_or_path,
                    unet=unwrapped_unet,
                    vae= vae,
                    scheduler=noise_scheduler,
                    tokenizer=tokenizer,
                    tokenizer_2=tokenizer_2,
                    text_encoder=text_encoder,
                    text_encoder_2=text_encoder_2,
                    image_encoder=image_encoder,
                    unet_encoder=unet_encoder,
                    torch_dtype=torch.float16,
                    add_watermarker=False,
                    safety_checker=None,
                )
                save_path = os.path.join(args.output_dir, f"checkpoint-{global_step}")
                pipeline.save_pretrained(save_path)
                del pipeline

Why isn't the image_proj_model saved here?
I also did not see the use of image_proj_model in the inference.py code......

**Comment by luxiaolili** [comment]
> if global_step % args.checkpointing_epoch == 0: if accelerator.is_main_process: # _before_ saving state, check if this save would set us over the `checkpoints_total_limit` unwrapped_unet = accelerator.unwrap_model( unet, keep_fp32_wrapper=True ) pipeline = TryonPipeline.from_pretrained( args.pretrained_model_name_or_path, unet=unwrapped_unet, vae= vae, scheduler=noise_scheduler, tokenizer=tokenizer, tokenizer_2=tokenizer_2, text_encoder=text_encoder, text_encoder_2=text_encoder_2, image_encoder=image_encoder, unet_encoder=unet_encoder, torch_dtype=torch.float16, add_watermarker=False, safety_checker=None, ) save_path = os.path.join(args.output_dir, f"checkpoint-{global_step}") pipeline.save_pretrained(save_path) del pipeline
> 
> Why isn't the image_proj_model saved here? I also did not see the use of image_proj_model in the inference.py code......

image_proj_model save in  tryon unet'encoder_hid_proj, so save the unet, also save the image_proj_model . You can read the code of unet_hacked_tryon.py  line 475
